src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtLocalAuthoritiesService.java

```diff
@@ -0,0 +1,83 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.CourtLocalAuthority;
+import uk.gov.hmcts.dts.fact.model.admin.LocalAuthority;
+import uk.gov.hmcts.dts.fact.repositories.CourtLocalAuthorityRepository;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+import uk.gov.hmcts.dts.fact.repositories.LocalAuthorityRepository;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class AdminCourtLocalAuthoritiesService {
+
+    private final LocalAuthorityRepository localAuthorityRepository;
+    private final CourtRepository courtRepository;
+    private final CourtLocalAuthorityRepository courtLocalAuthorityRepository;
+
+
+    @Autowired
+    public AdminCourtLocalAuthoritiesService(final LocalAuthorityRepository localAuthorityRepository, final CourtRepository courtRepository, final CourtLocalAuthorityRepository courtLocalAuthorityRepository) {
+        this.localAuthorityRepository = localAuthorityRepository;
+        this.courtRepository = courtRepository;
+        this.courtLocalAuthorityRepository = courtLocalAuthorityRepository;
+    }
+
+    public List<LocalAuthority> getAllLocalAuthorities() {
+        return localAuthorityRepository.findAll()
+            .stream()
+            .map(LocalAuthority::new)
+            .collect(toList());
+    }
+
+    public List<CourtLocalAuthority> getCourtLocalAuthoritiesBySlug(final String slug) {
+
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+
+        return courtLocalAuthorityRepository.findByCourtId(courtEntity.getId())
+                .stream()
+                .map(CourtLocalAuthority::new)
+                .collect(toList());
+
+    }
+
+    @Transactional(rollbackFor = {RuntimeException.class})
+    public List<CourtLocalAuthority> updateCourtLocalAuthority(final String slug, final List<CourtLocalAuthority> localAuthorities) {
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+        return saveNewCourtLocalAuthorities(courtEntity, localAuthorities);
+
+    }
+
+    protected List<CourtLocalAuthority> saveNewCourtLocalAuthorities(final Court courtEntity, final List<CourtLocalAuthority> localAuthorities) {
+
+        final List<uk.gov.hmcts.dts.fact.entity.CourtLocalAuthority> courtLocalAuthorityEntities = getNewCourtLocalAuthority(localAuthorities);
+        final List<uk.gov.hmcts.dts.fact.entity.CourtLocalAuthority> existingCourtLocalAuthorities = courtLocalAuthorityRepository.findByCourtId(courtEntity.getId());
+
+        //delete existing Court Local Authorities
+        courtLocalAuthorityRepository.deleteAll(existingCourtLocalAuthorities);
```

Comment:
Think there could be a problem here. When the admin portal send a request to the backend, it is for a single court and a single area of law. Here we are wiping out the information for that court for all area of law.

Reply:
refactor and implemented consideration for Area of Law when updating and getting local authorities